numba/tests/test_smart_array.py

```diff
@@ -57,8 +59,13 @@ def test_ufunc(self):
         cfunc = jit(nopython=True)(npyufunc_usecase)
         aa = cfunc(a)
         self.assertIsInstance(aa, SmartArray)
-        self.assertPreciseEqual(aa.host(), np.cos(np.sin(a.host())))
+        self.assertPreciseEqual(aa.get('host'), np.cos(np.sin(a.get('host'))))
 
+    def test_astype(self):
+        a = SmartArray(np.int32([42, 8, -5]))
+        aa = a.astype(np.float64)
+        self.assertIsInstance(aa, SmartArray)
+        self.assertPreciseEqual(aa.get('host'), a.get('host').astype(np.float64))
```

Comment:
Perhaps check the `dtype` is correct through comparison to a hardcoded dtype, else this is just checking that the function is working/broken in the same way?